### PR TITLE
Arguments

### PR DESCRIPTION
**Pull request by linted**
Added argument parsing to server and client. This doesn't cover every use case, and there may be issues if the user puts in a port number outside of the range of valid ports. It will throw an error in the code, not in the argparse so it won't be as pretty of an output. We can revisit it in a later feature if someone wants to write a custom type class for port, However i think that is out of scope for this issue.

### CODE REVIEW

**Comment by linted** [comment]
I have no idea what the heck happened to this branch. Honestly the best solution is to destroy this branch with fire and copy paste the code into a different new branch. @msmith533 can you help me with this after you review the code. Sorry to give you such a hard one to recover!

**Comment by linted** [comment]
@msmith533 I fixed everything up. This is clean to merge after it is reviewed.

File: server/server.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
 	import socketserver
 	import time
 	import sqlite3
+	import argparse
+	from ipaddress import ip_address
+	from sys import argv
```
Comment (calebmckay) [inline server/server.py]: Is _from sys import argv_ needed? It doesn't exist in client.py, and I don't see any real differences that would need it to be imported.

File: server/server.py [hunk 2]
```diff
@@ -48,15 +51,35 @@ def handle(self):
...
-	with open(path+ '/python.pub', 'w') as publicKey:
-		publicKey.write(key.publickey.exportKey('PEM'))
+	with open(path+ '/python.pub', 'wb') as publicKey:
+		publicKey.write(key.publickey().exportKey('PEM'))
+
+def parseArgs():
+	'''Parses args using the argparse lib'''
+	parser = argparse.ArgumentParser(description='Location logging server')
+
+	parser.add_argument('-g', '--generate-keys', metavar='PATH', type=str)
```
Comment (calebmckay) [inline server/server.py:65]: We probably want _nargs=1_ here, since we're only accepting one keygen path.

File: server/server.py (hunk 1 above)
Comment (linted) [reply server/server.py]: your right, I at one point removed that but this branch got screwed up (you can see it in the graphs->network)

File: server/server.py (hunk 2 above)
Comment (linted) [reply server/server.py:65]: actually we want the opposite. I had added nargs when I was toying with the idea of listening on multiple ports/IPs. With nargs the option is returned as a list, which seems weird for a single item. changes have been pushed to remove nargs.

Comment (calebmckay) [reply server/server.py:65]: Ah, indeed. I just saw this in the [argparse documentation](https://docs.python.org/3/library/argparse.html#nargs): 

> Note that nargs=1 produces a list of one item. This is different from the default, in which the item is produced by itself.

**Review by calebmckay — APPROVED** [review]

File: server/server.py (hunk 2 above)
Comment (calebmckay) [reply server/server.py:65]: If that's the case, you should probably also reflect this in client.py.

**Pull request merged**